Doc/reference/expressions.rst

```diff
@@ -823,12 +823,18 @@ An attribute reference is a primary followed by a period and a name:
 
 The primary must evaluate to an object of a type that supports attribute
 references, which most objects do.  This object is then asked to produce the
-attribute whose name is the identifier.  This production can be customized by
-overriding the :meth:`__getattr__` method.  If this attribute is not available,
-the exception :exc:`AttributeError` is raised.  Otherwise, the type and value of
-the object produced is determined by the object.  Multiple evaluations of the
-same attribute reference may yield different objects.
-
+attribute whose name is the identifier. The type and value produced is
+determined by the object.  Multiple evaluations of the same attribute
+reference may yield different objects.
+
+This production can be customized by overriding the
+:meth:`~object.__getattribute__` method or the :meth:`~object.__getattr__`
+method.  The :meth:`__getattribute__` method is called first and either
```

Comment:
Putting a `!` explicitly tells Sphinx that you don't expect it to add a link here, which means that Sphinx doesn't emit a warning here even if you enable Sphinx's nitpicky mode. That, in turn, helps out with the long-term project to fix all Sphinx warnings across CPython's docs (#101100 -- eventually we'll want to turn them into errors in CI)

```suggestion
method.  The :meth:`!__getattribute__` method is called first and either
```

Reply:
I do want the link that was generated.